### PR TITLE
Doxygen linear and NR strategy

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by msandre** [comment]
This is really good for the core :+1:

**Comment by loumalouomega** [comment]
(then approve me wink, wink)

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h [hunk 1]
```diff
@@ -191,121 +179,147 @@ class ResidualBasedLinearStrategy
...
+     * @param pNewBuilderAndSolver The pointer to the builder and solver considered
+     */
     void SetBuilderAndSolver(typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver)
     {
         mpBuilderAndSolver = pNewBuilderAndSolver;
     };
 
+    /**
+     * @brief Get method for the builder and solver
+     * @return mpBuilderAndSolver: The pointer to the builder and solver considered
```
Comment (msandre) [inline kratos/solving_strategies/strategies/residualbased_linear_strategy.h:257]: Is there a reason for the : here?

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h [hunk 2]
```diff
@@ -712,12 +711,15 @@ class ResidualBasedLinearStrategy
...
+        if (BaseType::GetEchoLevel() > 2 && rank == 0)
+            KRATOS_INFO("Exiting InitializeSolutionStep") << "Exiting the InitializeSolutionStep of the ResidualBasedLinearStrategy" << std::endl;
 
         KRATOS_CATCH("")
     }
 
-    //**********************************************************************
-    //**********************************************************************
+    /**
+     * @brief This method prints information after reach the max number of interations
```
Comment (msandre) [inline kratos/solving_strategies/strategies/residualbased_linear_strategy.h:721]: iterations has a typo.

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h [hunk 3]
```diff
@@ -177,9 +171,20 @@ class ResidualBasedNewtonRaphsonStrategy
...
-
+    
+    /**
+     * Constructor specifying the builder and solver
+     * @param rModelPart The model part of the problem
+     * @param pScheme The integration scheme
+     * @param pNewLinearSolver The linear solver employed
+     * @param pNewConvergenceCriteria The convergence criteria employed
+     * @param pNewBuilderAndSolver The builder and solver employed
+     * @param CalculateReactionFlag The flag for the reaction calculation
```
Comment (msandre) [inline kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: MaxIterations is missing.

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h [hunk 4]
```diff
@@ -116,59 +104,65 @@ class ResidualBasedNewtonRaphsonStrategy
...
 
-    /** 
-     * Constructor.
+    /**
+     * Default constructor 
+     * @param rModelPart The model part of the problem
+     * @param pScheme The integration scheme
+     * @param pNewLinearSolver The linear solver employed
+     * @param pNewConvergenceCriteria The convergence criteria employed
+     * @param CalculateReactions The flag for the reaction calculation
```
Comment (msandre) [inline kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: MaxIterations is missing.

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h [hunk 5]
```diff
@@ -51,26 +51,13 @@ namespace Kratos
...
-
-//URL[Extended documentation doc]{ extended_documentation/no_ext_doc.doc}
-
-//URL[Extended documentation ps]{ extended_documentation/no_ext_doc.ps}
-
+/**   
+ * @class ResidualBasedNewtonRaphsonStrategy
+ * @ingroup KratosCore
+ * @brief This is the base Newton Raphson strategy
+ * @details This strategy iterates until the convergence is achieved (or the maximum number of iterations is suppased) using a Newton Raphson algorithm
```
Comment (msandre) [inline kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: surpassed has a typo.

**Review by msandre — CHANGES_REQUESTED** [review]
Some minor changes.

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_linear_strategy.h:257]: Sorry

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_linear_strategy.h:721]: I am uhman

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h (hunk 3 above)
Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: True

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h (hunk 4 above)
Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: True

File: kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h (hunk 5 above)
Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h]: Okidoki

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/solving_strategies/strategies/residualbased_linear_strategy.h:257]: No, it is Ok for @return

**Comment by loumalouomega** [comment]
Done

**Comment by msandre** [comment]
Thx, but the spelling, can you fix it?

**Review by msandre — APPROVED** [review]

**Comment by msandre** [comment]
Nevermind I'm confused.

**Pull request merged**